src/ivoc/xmenu.cpp

```diff
@@ -1062,30 +1054,28 @@ void HocUpdateItem::check_pointer(void*, int) {}
 void HocUpdateItem::data_path(HocDataPaths*, bool) {}
 
 // ones that get updated on every doEvents()
-HocUpdateItemList* HocPanel::update_list_;
+std::vector<HocUpdateItem*>* HocPanel::update_list_;
 
 void HocPanel::keep_updated() {
     static int cnt = 0;
     if (update_list_ && (++cnt % 10 == 0)) {
-        long i, cnt = update_list_->count();
+        long i, cnt = update_list_->size();
         if (cnt)
-            for (i = 0; i < cnt; ++i) {
-                update_list_->item(i)->update_hoc_item();
+            for (auto& item: *update_list_) {
+                item->update_hoc_item();
             }
     }
 }
 void HocPanel::keep_updated(HocUpdateItem* hui, bool add) {
     if (!update_list_) {
-        update_list_ = new HocUpdateItemList();
+        update_list_ = new std::vector<HocUpdateItem*>();
     }
     if (add) {
-        update_list_->append(hui);
+        update_list_->push_back(hui);
     } else {
-        for (long i = 0; i < update_list_->count(); ++i) {
-            if (update_list_->item(i) == hui) {
-                update_list_->remove(i);
-                break;
-            }
+        auto it = std::find(update_list_->begin(), update_list_->end(), hui);
+        if (it != update_list_->end()) {
+            update_list_->erase(it);
```

Comment:
Possibly non-ideal pattern for `std::vector`.

Reply:
why?